src/components/ModelGroupToggle/ModelGroupToggle.js

```diff
@@ -1,11 +1,21 @@
-import React from "react";
+import React, { useEffect } from "react";
+import { useHistory } from "react-router-dom";
 import classNames from "classnames";
 
 import "./_model-group-toggle.scss";
 
 const buttons = ["status", "cloud", "owner"];
 
 const ModelGroupToggle = ({ groupedBy, setGroupedBy }) => {
+  const history = useHistory();
+
+  useEffect(() => {
+    history.push({
+      pathname: "/models",
+      search: `?groupby=${groupedBy}`
```

Comment:
I think this will have the same problem...what happens if your url is `/models?groupby=owner&filterby=gce` ? 

What you'll need to do is fetch the query param, update the specific element you're interested in then push the updated value... You may find something like https://www.npmjs.com/package/query-string to be helpful - note I did not inspect that package, just adding it as a push off point.

We also don't need to set a groupby query param for the default grouping

Reply:
You probably just been to prepend the existing search query?